test/suites/integration/bsp/debt-collection.test.ts

```diff
@@ -181,27 +152,533 @@ describeBspNet(
+      let usersWithDebtResult = await bspApi.call.paymentStreamsApi.getUsersWithDebtOverThreshold(
+        ShConsts.DUMMY_BSP_ID,
+        0
+      );
+      assert(usersWithDebtResult.isOk);
+      assert(usersWithDebtResult.asOk.length === 1);
+      assert(usersWithDebtResult.asOk[0].toString() === userAddress);
+
+      // Get the last chargeable info of the dummy BSP before proof submission
+      let lastChargeableInfo = await userApi.query.paymentStreams.lastChargeableInfo(
+        ShConsts.DUMMY_BSP_ID
+      );
+
+      // Seal one more block with the pending extrinsics.
+      await userApi.sealBlock();
+
+      // Get the current price of storage from the runtime, the new stream deposit and the ED
+      const currentPriceOfStorage = await userApi.query.paymentStreams.currentPricePerUnitPerTick();
+      const newStreamDeposit = await userApi.consts.paymentStreams.newStreamDeposit;
+      const existentialDeposit = await userApi.consts.balances.existentialDeposit;
+
+      // Get the current free balance of the user
+      const freeBalance = (await userApi.query.system.account(userAddress)).data.free;
+
+      // To make the user insolvent, we need to update the payment stream with a very high amount
+      // and advance new stream deposit blocks
+      // To do this, the new amount provided should be equal to the free balance of the user divided by
+      // the current price of storage multiplied by the new stream deposit
+      const newAmountProvidedForInsolvency = freeBalance
+        .div(currentPriceOfStorage.mul(newStreamDeposit))
+        .sub(existentialDeposit);
+
+      // Make the user insolvent by updating the payment stream with a very high amount
+      const updateDynamicRatePaymentStreamResult = await userApi.sealBlock(
+        userApi.tx.sudo.sudo(
+          userApi.tx.paymentStreams.updateDynamicRatePaymentStream(
+            ShConsts.DUMMY_BSP_ID,
+            userAddress,
+            newAmountProvidedForInsolvency
+          )
+        )
+      );
+      const { extSuccess } = updateDynamicRatePaymentStreamResult;
+      strictEqual(extSuccess, true, "Extrinsic should be successful");
+
+      // Assert that event dynamic-rate payment stream creation was emitted
+      userApi.assertEvent(
+        "paymentStreams",
+        "DynamicRatePaymentStreamUpdated",
+        updateDynamicRatePaymentStreamResult.events
+      );
+      // Get the on-chain payment stream information
+      const [userAccount, providerId, newAmountProvided] = fetchEventData(
+        userApi.events.paymentStreams.DynamicRatePaymentStreamUpdated,
+        await userApi.query.system.events()
+      );
+      // Assert that the information on-chain is correct
+      strictEqual(userAccount.toString(), userAddress);
+      strictEqual(providerId.toString(), ShConsts.DUMMY_BSP_ID.toString());
+      strictEqual(newAmountProvided.toNumber(), newAmountProvidedForInsolvency.toNumber());
+
+      ///////** INITIAL PROOF SUBMISSION **///////
```

Comment:
Suggestion: separate these parts into different `it` blocks.

Reply:
Done